### PR TITLE
cli: add submodules using git interface

### PR DESCRIPTION
**Pull request by michaelsevilla**
- specialized gitmodules file tells popper to 'git submodule add'
- experiment add does not copy submodules with cp -r
- for clarity we avoid using YAML and git-specific formats

Signed-off-by: Michael Sevilla <mikesevilla3@gmail.com>

### CODE REVIEW

**Comment by michaelsevilla** [comment]
Addresses #120. @ivotron PTAL 

Need some feedback.

File: popper/experiment.go [hunk 1]
```diff
@@ -63,6 +65,64 @@ func listTemplates(templateType string) {
...
+	s := bufio.NewScanner(f)
+	s.Split(bufio.ScanLines)
+
+	// gs (gitmodule settings): nested map that maps keys of type string to maps
+	var gs map[string]map[string]string = make(map[string]map[string]string)
+
+	// g (gitmodule): the current module we are parsing
+	var g string
+
+	// parse gitmodule settings (this is a hack)
```
Comment (michaelsevilla) [inline popper/experiment.go]: This parsing is hideous. @ivotron: what do you think? Can we make this cleaner?

File: popper/experiment.go [hunk 2]
```diff
@@ -63,6 +65,64 @@ func listTemplates(templateType string) {
...
+			gs[g] = make(map[string]string)
+		} else {
+			// we are parsing a gitmodule attribute
+			words := strings.Split(line, "=")
+			key := strings.TrimSpace(words[0])
+			val := strings.TrimSpace(words[1])
+
+			// save attribute for current gitmodule
+			switch key {
+			case "path":
```
Comment (michaelsevilla) [inline popper/experiment.go]: Potential to overwrite attribute if the user creates a malformed `.gitmodules` file for the template.

File: popper/experiment.go [hunk 3]
```diff
@@ -63,6 +65,64 @@ func listTemplates(templateType string) {
...
+			case "url":
+				gs[g]["url"] = val
+			case "hash":
+				gs[g]["hash"] = val
+			}
+		}
+	}
+
+	// use git to add submodule(s)
+	for _, g := range gs {
```
Comment (michaelsevilla) [inline popper/experiment.go]: Could fail in the middle, in which case we have checked out master and we aren't on the correct hash/branch.

File: popper/experiment.go [hunk 4]
```diff
@@ -63,6 +65,64 @@ func listTemplates(templateType string) {
...
+		path:= folder + "/" + g["path"]
+		if err := sh.Command("git", "submodule", "add", g["url"], path).Run(); err != nil {
+			log.Fatalln(err)
+		}
+		if err := sh.Command("git", "checkout", "-b", g["hash"], g["hash"], sh.Dir(path)).Run(); err != nil {
```
Comment (michaelsevilla) [inline popper/experiment.go]: Could fail in the middle, in which case we have checked out master and we aren't on the correct hash/branch.

**Comment by ivotron** [comment]
@michaelsevilla I think I finally found a practical solution: popper repos shouldn't have submodules. If you need a submodule, it probably is an indication that whatever you're submoduling needs its own package (e.g. docker image, pip package, cli interface, etc..). In the case of the ceph baseliner, I feel like it's sufficiently complex that you can put it on its own docker image. what do you think?

**Comment by ivotron** [comment]
subsumed by #70

**Pull request closed without merging**